### PR TITLE
gs1.1: chore: add go gossipsub tests

### PR DESCRIPTION
**Pull request by wemeetagain**
Add go gossipsub tests from: https://github.com/libp2p/go-libp2p-pubsub/blob/master/gossipsub_test.go
A few irrelevant tests have been left out.

A few bugs have been found in the course of running these tests, see #115
A few tests are flaky, a few are failing (after #115)

### CODE REVIEW

File: test/go-gossipsub.spec.js [hunk 1]
```diff
@@ -0,0 +1,1115 @@
...
+
+    // wait for heartbeats
+    await delay(2000)
+
+    psubs.slice(1).forEach(ps => ps.subscribe(topic))
+
+    // wait for heartbeats
+    await delay(2000)
+
+    for (let i = 0; i < 100; i++) {
```
Comment (vasco-santos) [inline test/go-gossipsub.spec.js:217]: The loops are equal. We should put them inside a function

File: test/go-gossipsub.spec.js [hunk 2]
```diff
@@ -0,0 +1,1115 @@
...
+      const msg = Buffer.from(`${i} its not a flooooood ${i}`)
+      const owner = Math.floor(Math.random() * psubs.length)
+      const results = Promise.all(
+        psubs
+          .filter((psub, j) => j !== owner)
+          .map(checkReceivedMessage(topic, msg, owner, i))
+      )
+      await psubs[owner].publish(topic, msg)
```
[the remark this answers is not in the document]
Comment (wemeetagain) [reply test/go-gossipsub.spec.js:78]: :+1: In most of the tests, I think we can do it.
In a few, we explicitly pause after each publish.

File: test/go-gossipsub.spec.js [hunk 3]
```diff
@@ -0,0 +1,1115 @@
...
+      number: 20,
+      options: { scoreParams: { IPColocationFactorThreshold: 20 } }
+    })
+    const topic = 'foobar'
+    psubs.forEach(ps => ps.subscribe(topic))
+
+    await sparseConnect(psubs)
+
+    // wait for heartbeats to build mesh
+    await delay(2000)
```
[the remark this answers is not in the document]
Comment (wemeetagain) [reply test/go-gossipsub.spec.js:68]: Yea, we can track each node's "gossipsub:heartbeat" event

**Review by vasco-santos — APPROVED** [review]
LGTM 🎉

**Pull request merged**